src/nvim/hardcopy.c

```diff
@@ -728,13 +730,10 @@ void ex_hardcopy(exarg_T *eap)
           if (got_int || settings.user_abort)
             goto print_fail;
 
-          sprintf((char *)IObuff, _("Printing page %d (%d%%)"),
-              page_count + 1 + side,
-              prtpos.bytes_printed > 1000000
-              ? (int)(prtpos.bytes_printed /
-                      (bytes_to_print / 100))
-              : (int)((prtpos.bytes_printed * 100)
-                      / bytes_to_print));
+          assert(prtpos.bytes_printed * 100 > prtpos.bytes_printed);
```

Comment:
Right. But I don't really get the initial code either. Is it trying to avoid overflows?